### ISSUE TITLE
G-API: revise generated documentation

### ISSUE THREAD

**Issue by alalek** [opencv/opencv#18963]
There is link on generated [gapi namespace](https://docs.opencv.org/master/d4/d1c/namespacecv_1_1gapi.html) documentation (content will be updated after fixes).

Issues:
- [ ] broken layout after/in `BGR2Gray` entry
- [ ] navigation header doesn't look correct too (grouping issues):
    `G-API framework » G-API Graph Compilation ArgumentsG-API framework » G-API Data Types » | G-API Metadata Descriptors`
- [ ] perhaps we need to ignore in documentation entries with `std::enable_if< std::is_same< T, GCompileArgs >::value, GCompileArgs >::type` (or move it into nested group through [`@ingroup`](https://www.doxygen.nl/manual/grouping.html))

**Comment by alalek** [comment]
One more broken documentation [page](https://docs.opencv.org/master/dc/d38/group__gapi__colorconvert.html).

**Comment by legercheap** [comment]
http://finance.losaltos.com/camedia.losaltos/news/read/40736255
https://is.gd/TD5hnK
https://is.gd/S6S6NS
https://is.gd/pqKhsx
https://is.gd/Wo7OPc
https://is.gd/VPTqQh
http://quotes.valueinvestingnews.com/fatpitch.valueinvestingnews/news/read/40736255
https://bit.ly/2IMdAri
https://bit.ly/345A11Y
https://bit.ly/3m9zpyt
https://bit.ly/37cRMhI
https://bit.ly/3a4vv7E
https://lifestyle.asialogue.com/news/the-history-for-the-herve-leger/0268750
https://tinyurl.com/yyylxwdk
https://tinyurl.com/yy4vm2gc
https://tinyurl.com/y4brw734
https://tech.asialogue.com/news/the-history-for-the-herve-leger/0268750
https://cutt.ly/khPIipc
https://cutt.ly/uhPOrhu
https://cutt.ly/xhPOr49
https://cutt.ly/chPOtz0
https://cutt.ly/chPOu54
https://entertainment.frontalreport.com/news/the-history-for-the-herve-leger/0268750
https://rebrand.ly/vyq66
https://rebrand.ly/jkojr
https://rebrand.ly/9s059
https://rebrand.ly/sny0q
https://rebrand.ly/a36qm
http://b.link/7xfxl
http://b.link/ktywc
http://b.link/fugkt
http://b.link/8kpzx
http://b.link/fgad5
http://tiny.cc/lol6tz

**Comment by dmatveev** [comment]
@alalek  the BGR2Gray doxygen header (where the formatting breaks) hasn't changed since 2018 (its inception). Not sure where that problem is

**Comment by alalek** [comment]
I have no idea what exactly explodes this problem.

`@brief` section is very large. They must be very simple one-liners. All details including formulas (!) must go into `@details`.

Try to insert empty new-line [here](https://github.com/opencv/opencv/blame/fcdd69fd972d96cfc7861948a4b2bc5d20d3cf42/modules/gapi/include/opencv2/gapi/imgproc.hpp#L1373).

**Comment by dmatveev** [comment]
Yes, it seems the resulting doxygen header becomes too complex and it simply breaks due to formula previews on `@brief`. Separating this fixes the issue.

**Comment by dmatveev** [comment]
> perhaps we need to ignore in documentation entries with std::enable_if< std::is_same< T, GCompileArgs >::value, GCompileArgs >::type (or move it into nested group through @ingroup)

Not sure how it helps... Those `enable_if` methods shouldn't be ignored as they part of the API, and need to be documented individually. The current-generated doc fully reflects the specifics of those methods:

![image](https://user-images.githubusercontent.com/144187/102266216-30363500-3f29-11eb-87c5-32056345105d.png)

**Comment by dmatveev** [comment]
> navigation header doesn't look correct too (grouping issues):

The navigation header is auto-generated for a namespace, again. We have different documentation groups within the same namespace. Not sure what is the right solution here. There is the same problem with `cv::`: https://docs.opencv.org/master/d2/d75/namespacecv.html

**Comment by asmorkalov** [comment]
@TolyaTalamanov Could you do it for 4.8.0 release?
